### ISSUE TITLE
Improve typing of `manifest` config

### ISSUE THREAD

**Issue by aklinker1** [wxt-dev/wxt#521]
### Feature Request

Instead of using `webextension-polyfill`'s `Manifest.WebExtensionManifest` type, implement our own that works for both Chrome and firefox, as well as MV2 and MV3.

#### Is your feature request related to a bug?

N/A

### What are the alternatives?

TS ignoring things everywhere.

### Additional context

Realized the types weren't good enough in #518

**Comment by pavitra-infocusp** [comment]
Missing `manifest.key` property type on `wxt.config.ts`. It works though.

```
Object literal may only specify known properties, and 'key' does not exist in type 'Partial<Omit<WebExtensionManifest, "background" | "chrome_url_overrides" | "devtools_page" | "manifest_version" | "options_page" | "options_ui" | "sandbox">> | Promise<...> | UserManifestFn'.ts(2353)
index-l43sonfW.d.ts(227, 5): The expected type comes from property 'manifest' which is declared here on type 'UserConfig'
(property) key: string
```

**Comment by trinhcanhphuc** [comment]
any solution for this issue ? :(

**Comment by aklinker1** [comment]
Update, after adding support for `@types/chrome` via `extensionApi: "chrome"`, I'm planning on migrating the manifest type to `@types/chrome` as well. Need to review it's types to see how well it supports Firefox-only fields.

**Comment by aklinker1** [comment]
Released in [`v0.19.9`](https://github.com/wxt-dev/wxt/releases/tag/wxt-v0.19.9)
